solr/core/src/test/org/apache/solr/cloud/MultiThreadedOCPTest.java

```diff
@@ -79,40 +78,57 @@ private void testFillWorkQueue() throws Exception {
             QUEUE_OPERATION, MOCK_COLL_TASK.toLower(),
             ASYNC, String.valueOf(i),
 
-            "sleep", (i == 0 ? "1000" : "1") //first task waits for 1 second, and thus blocking
-            // all other tasks. Subsequent tasks only wait for 1ms
+            // third task waits for a long time, and thus blocks the queue for all other tasks for A_COLL.
+            // Subsequent tasks as well as the first two only wait for 1ms
+            "sleep", (i == 2 ? "10000" : "1")
         )));
         log.info("MOCK task added {}", i);
+      }
 
+      // Wait until we see the first two A_COLL tasks getting processed
+      boolean acoll0done = false, acoll1done = false;
```

Comment:
I found this section a little bit hard to read, would prefer something more verbose but potentially easier to understand at a glance.

Reply:
Extracted two methods and simplified flow.